core/src/main/java/nl/nn/adapterframework/jdbc/migration/Migrator.java

```diff
@@ -41,19 +42,16 @@ public Migrator() {
 
 	@Override
 	public void configure() throws ConfigurationException {
-		configure(null, null, null);
+		throw new IllegalStateException("No configuration is specified!");
 	}
 
-	public void configure(String configurationName) throws ConfigurationException {
-		configure(configurationName, null, null);
+	public void configure(Configuration configuration, ClassLoader classLoader) throws ConfigurationException {
+		configure(configuration, classLoader, null);
 	}
 
-	public void configure(String configurationName, ClassLoader classLoader) throws ConfigurationException {
-		configure(configurationName, classLoader, null);
-	}
-
-	public synchronized void configure(String configurationName, ClassLoader classLoader, String changeLogFile) throws ConfigurationException {
-		setName("JdbcMigrator for configuration["+configurationName+"]");
+	public synchronized void configure(Configuration configuration, ClassLoader classLoader, String changeLogFile) throws ConfigurationException {
+		setName("JdbcMigrator for configuration["+ configuration.getName() +"]");
+		super.configure();
```

Comment:
You don't need to specify the `ClassLoader` as you can retrieve this from the `Configuration`.